test/evp_pkey_provided_test.c

```diff
@@ -1074,7 +1080,10 @@ static int test_fromdata_dh_named_group(void)
         ret = test_print_key_using_pem("DH", pk)
               && test_print_key_using_encoder("DH", pk);
 
-        if (!ret || !TEST_ptr(dup_pk = EVP_PKEY_dup(pk)))
+        if (!ret || dup_pk != NULL)
+            break;
```

Comment:
does using `!ret` go against our coding-style?

I don't necessarily think it should be changed (it is used in quite a few places in these files).

Reply:
Nope, if `ret` is boolean (which it is), it is fine to use `!ret`.